src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2932,7 +2932,7 @@ protected override IPointerTypeSymbol CommonCreatePointerTypeSymbol(ITypeSymbol
 
             if (!memberIsReadOnly.IsDefault && memberIsReadOnly.Any(v => !v))
             {
-                throw new ArgumentException($"Non-ReadOnly members are not supported in C# anonymous types.");
```

Comment:
This is actually unnecessary -- our policy is that exceptions don't need to be localized unless we know they're going to be turned into a diagnostic.